src/runtime/runtime.cs

```diff
@@ -310,99 +331,92 @@ internal static void Initialize(bool initSigs = false)
-            IntPtr path = PySys_GetObject("path");
-            IntPtr item = PyString_FromString(rtdir);
-            PyList_Append(path, item);
-            XDecref(item);
-            AssemblyManager.UpdatePath();
-        }
-
-        /// <summary>
-        /// Initializes the data about platforms.
-        ///
-        /// This must be the last step when initializing the runtime:
-        /// GetManagedString needs to have the cached values for types.
-        /// But it must run before initializing anything outside the runtime
-        /// because those rely on the platform data.
-        /// </summary>
-        private static void InitializePlatformData()
-        {
-            IntPtr op;
-            IntPtr fn;
-            IntPtr platformModule = PyImport_ImportModule("platform");
-            IntPtr emptyTuple = PyTuple_New(0);
+            var mode = ShutdownMode;
+#if !NETSTANDARD
+            if (mode == ShutdownMode.Reload)
+            {
+                RuntimeData.Stash();
+            }
+#endif
+            AssemblyManager.Shutdown();
+            ImportHook.Shutdown();
 
-            fn = PyObject_GetAttrString(platformModule, "system");
-            op = PyObject_Call(fn, emptyTuple, IntPtr.Zero);
-            OperatingSystemName = GetManagedString(op);
-            XDecref(op);
-            XDecref(fn);
+            ClearClrModules();
+            RemoveClrRootModule();
 
-            fn = PyObject_GetAttrString(platformModule, "machine");
-            op = PyObject_Call(fn, emptyTuple, IntPtr.Zero);
-            MachineName = GetManagedString(op);
-            XDecref(op);
-            XDecref(fn);
+            MoveClrInstancesOnwershipToPython();
+            ClassManager.RemoveClasses();
+            TypeManager.RemoveTypes();
 
-            XDecref(emptyTuple);
-            XDecref(platformModule);
+            MetaType.Release();
+            PyCLRMetaType = IntPtr.Zero;
 
-            // Now convert the strings into enum values so we can do switch
-            // statements rather than constant parsing.
-            OperatingSystemType OSType;
-            if (!OperatingSystemTypeMapping.TryGetValue(OperatingSystemName, out OSType))
-            {
-                OSType = OperatingSystemType.Other;
-            }
-            OperatingSystem = OSType;
+            Exceptions.Shutdown();
+            Finalizer.Shutdown();
 
-            MachineType MType;
-            if (!MachineTypeMapping.TryGetValue(MachineName.ToLower(), out MType))
+            if (mode != ShutdownMode.Normal)
             {
-                MType = MachineType.Other;
+                PyGC_Collect();
+                if (mode == ShutdownMode.Soft)
+                {
+                    RuntimeState.Restore();
+                }
+                ResetPyMembers();
+                GC.Collect();
+                try
+                {
+                    GC.WaitForFullGCComplete();
```

Comment:
I don't think `WaitForFullGCComplete` is the right method to use here. See https://stackoverflow.com/questions/32219440/waitforfullgccomplete-vs-waitforpendingfinalizers-collect#32219800

Perhaps you meant `WaitForPendingFinalizers`?